warehouse/query-core/src/main/java/datawave/query/discovery/DiscoveryTransformer.java

```diff
@@ -17,22 +13,25 @@
 import datawave.webservice.query.logic.BaseQueryLogicTransformer;
 import datawave.webservice.query.result.event.EventBase;
 import datawave.webservice.query.result.event.FieldBase;
-import datawave.webservice.query.result.event.ResponseObjectFactory;
 import datawave.webservice.query.result.event.Metadata;
+import datawave.webservice.query.result.event.ResponseObjectFactory;
 import datawave.webservice.result.BaseQueryResponse;
 import datawave.webservice.result.EventQueryResponseBase;
-
 import org.apache.accumulo.core.security.ColumnVisibility;
 import org.apache.hadoop.io.Writable;
 
-import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 public class DiscoveryTransformer extends BaseQueryLogicTransformer<DiscoveredThing,EventBase> implements CacheableLogic {
     private List<String> variableFieldList = null;
     private BaseQueryLogic<DiscoveredThing> logic = null;
     private QueryModel myQueryModel = null;
     private MarkingFunctions markingFunctions;
     private ResponseObjectFactory responseObjectFactory;
+    private long queryExecutionForCurrentPageStartTime;
```

Comment:
Should this actually be located in the BaseQueryLogicTransformer?

Reply:
same comment on all other extensions of BaseQueryLogicTransformer below....

Reply:
Not sure what I was thinking.... After another look, I found that I could actually put it in `AbstractQueryLogicTransformer` (BaseQueryLogicTransformer extends AbstractQueryLogicTransformer and there are a few Transformers that extend AbstractQueryLogicTransformer directly) and I removed it from the children classes.